### ISSUE TITLE
Typed Arrays

### ISSUE THREAD

**Issue by neothemachine** [zloirock/core-js#90]
I wanted to use the slice function of TypedArray and saw there is no polyfill. I also couldn't find one somewhere else with a quick search. Is it possible to implement that? I'm not sure as it seems to be quite native and hard to emulate without actually making a copy of the array. But maybe (hopefully) I'm wrong.

**Comment by neothemachine** [comment]
Ah wait, it does actually make a copy. They say "shallow" copy, but typed arrays are of primitive types anyway so I don't really understand this distinction. Possibly for future types which can host references to more structured things. So, a polyfill should be easy then, basically a for loop copying elements into a new typed array of the same type and returning it.

**Comment by zloirock** [comment]
Currently, `core-js` not adds typed arrays polyfill. It's possible, but slow - getters / setters for each element in modern environment (but most missing, IIRC, only in IE9) and not complete in ES3 environment. You can see [this polyfill](https://github.com/inexorabletash/polyfill/blob/master/typedarray.js).

I'm thinking about adding only missing methods or, possible, adding complete polyfill, but not in near versions.

**Comment by neothemachine** [comment]
Given the fast advances of browsers I wouldn't waste time adding full typed arrays polyfills, but rather only add the missing functions of the latest browser versions. Like the slice method in my case. I see core-js more as a tool to be able to use the future now, not as something to support old browsers, but of course, there is some overlap.

**Comment by neothemachine** [comment]
For reference, this is my working polyfill, adapted from https://github.com/inexorabletash/polyfill/blob/master/typedarray.js:

```js
"use strict"

for (let TA of [Int8Array,Uint8Array,Int16Array,Uint16Array,Int32Array,Uint32Array,Float32Array,Float64Array]) {
  if (!TA.prototype.slice) {
    TA.prototype.slice = function (start, end) {
      var len = this.length
      var relativeStart = start
      var k = (relativeStart < 0) ? max(len + relativeStart, 0) : Math.min(relativeStart, len)
      var relativeEnd = (end === undefined) ? len : end
      var final = (relativeEnd < 0) ? max(len + relativeEnd, 0) : Math.min(relativeEnd, len)
      var count = final - k
      var c = this.constructor
      var a = new c(count)
      var n = 0
      while (k < final) {
        var kValue = this[k]
        a[n] = kValue
        ++k
        ++n
      }
      return a
    }
  }
}
```

Note that I couldn't use `Object.getPrototypeOf(Int8Array.prototype)` to get to the `%TypedArray%` constructor as described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray). Chrome just gave me back `Object`. So I worked around that by explicitly adding the function to each type.

**Comment by Yaffle** [comment]
@neothemachine  `Object.getPrototypeOf(this)`

**Comment by zloirock** [comment]
Available in [2.0](https://github.com/zloirock/core-js/releases/tag/v2.0.0).

**Comment by neothemachine** [comment]
:+1:
